### PR TITLE
[Enhancement] Reduce the number of calls to refresh the iceberg table

### PR DESCRIPTION
**Pull request by caneGuy**
## What type of PR is this：
- [ ] bugfix
- [ ] feature
- [x] enhancement
- [ ] refactor
- [ ] others

## Which issues of this PR fixes ：

Fixes #

## Problem Summary(Required) ：

1. No need to call refresh directly
2. Cache loaded table for query

## Checklist:

- [ ] I have added test cases for my bug fix or my new feature
- [ ] I have added user document for my new feature or new function

### CODE REVIEW

File: fe/fe-core/src/main/java/com/starrocks/sql/optimizer/transformer/RelationTransformer.java [hunk 1]
```diff
@@ -430,7 +430,9 @@ public LogicalPlan visitTable(TableRelation node, ExpressionMapping context) {
             scanOperator = new LogicalHiveScanOperator(node.getTable(), node.getTable().getType(),
                     colRefToColumnMetaMapBuilder.build(), columnMetaToColRefMap, Operator.DEFAULT_LIMIT, null);
         } else if (Table.TableType.ICEBERG.equals(node.getTable().getType())) {
-            ((IcebergTable) node.getTable()).refreshTable();
+            if (!((IcebergTable) node.getTable()).isCatalogTbl()) {
```
Comment (DorianZheng) [inline fe/fe-core/src/main/java/com/starrocks/sql/optimizer/transformer/RelationTransformer.java:433]: why only the non-catalog table should perform `refresh` ops

**Review by dirtysalt — DISMISSED** [review]

File: fe/fe-core/src/main/java/com/starrocks/sql/optimizer/transformer/RelationTransformer.java (hunk 1 above)
Comment (caneGuy) [reply fe/fe-core/src/main/java/com/starrocks/sql/optimizer/transformer/RelationTransformer.java:433]: catalog table will load each time when get table during query analyze phase

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:disappointed: **fail** : 7 / 10 (70.00%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/sql/optimizer/transformer/RelationTransformer.java|0|2|00.00%|[433, 434]|
|:large_blue_circle:|com/starrocks/catalog/IcebergTable.java|5|6|83.33%|[128]|
|:large_blue_circle:|com/starrocks/external/iceberg/IcebergUtil.java|2|2|100.00%|[]|

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:disappointed: **fail** : 7 / 10 (70.00%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/sql/optimizer/transformer/RelationTransformer.java|0|2|00.00%|[433, 434]|
|:large_blue_circle:|com/starrocks/catalog/IcebergTable.java|5|6|83.33%|[128]|
|:large_blue_circle:|com/starrocks/external/iceberg/IcebergUtil.java|2|2|100.00%|[]|

**Pull request closed without merging**